link/kprobe.go

```diff
@@ -191,6 +198,20 @@ func kprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions, ret bool) (*
 	}
 
 	if opts != nil {
+		if opts.RetprobeMaxActive < 0 {
+			return nil, fmt.Errorf("requires opts.RetprobeMaxActive >= 0: %w", errInvalidInput)
+		} else if opts.RetprobeMaxActive > 0 {
+			// The kretprobe's maxactive can be set on kernels 4.12 and up.
+			v, err := internal.KernelVersion()
+			if err == nil {
+				kv, _ := internal.NewVersion("4.12")
+				if v.Less(kv) {
+					args.retprobeMaxActive = 0
```

Comment:
Please remove the fallback based on kernel version or replace it with a full feature probe. Comparing kernel versions is not reliable.

Reply:
Tell me more, what do you have in mind? I'm not very familiar with tracefs myself.